pjlib/src/pj/ssl_sock_ossl.c

```diff
@@ -2436,15 +2449,25 @@ static pj_status_t ssl_read(pj_ssl_sock_t *ssock, void *data, int *size)
                 /* Reset SSL socket state, then return PJ_FALSE */
                 status = STATUS_FROM_SSL_ERR2("Read", ssock, size_,
                                               err, len);
+                pj_lock_release(ssock->write_mutex);
+                /* Unfortunately we can't hold the lock here to reset all the state.
+                  * We probably should though.
+                  *
+                  * TODO: if we get an SSL_ERROR_SSL or SSL_ERROR_SYSCALL, we need to not perform
```

Comment:
Since we're going to immediately address this in the other PR, I suppose this comment can be removed?

Reply:
Ok, just pushed a fix up that removes the TODO portion of the comment.